x/finality/keeper/msg_server.go

```diff
@@ -333,3 +333,56 @@ func (ms msgServer) validateActivationHeight(ctx sdk.Context, height uint64) (ui
 	}
 	return activationHeight, nil
 }
+
+// EquivocationEvidence handles the evidence of equivocation message sent from the finality gadget cw contract
+func (ms msgServer) EquivocationEvidence(goCtx context.Context, req *types.MsgEquivocationEvidence) (*types.MsgEquivocationEvidenceResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// validate the evidence
+	if err := validateEvidence(req); err != nil {
+		return &types.MsgEquivocationEvidenceResponse{}, err
+	}
```

Comment:
```suggestion
	// validate the evidence
	if err := req.ValidateBasics(); err != nil {
		return &types.MsgEquivocationEvidenceResponse{}, err
	}
```

Reply:
The helper function `validateEvidence` is created in `msg_server.go`, and there is the below comments for `ValidateBasic` in Cosmos SDK
```
// ValidateBasic is deprecated and now facultative.
// Prefer validating messages directly in the msg server.
```